app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -1,46 +1,87 @@
 package com.lightricks.feedexercise.ui.feed
 
+import android.app.Application
 import androidx.lifecycle.*
+import androidx.room.Room
 import com.lightricks.feedexercise.data.FeedItem
+import com.lightricks.feedexercise.data.FeedRepository
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.network.FeedApiService
 import com.lightricks.feedexercise.util.Event
-import java.lang.IllegalArgumentException
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
 
 /**
  * This view model manages the data for [FeedFragment].
  */
-open class FeedViewModel : ViewModel() {
+open class FeedViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val BASE_URL = "https://assets.swishvideoapp.com/"
     private val isLoading = MutableLiveData<Boolean>()
     private val isEmpty = MutableLiveData<Boolean>()
     private val feedItems = MediatorLiveData<List<FeedItem>>()
     private val networkErrorEvent = MutableLiveData<Event<String>>()
+    private val feedRepository = FeedRepository(
```

Comment:
`FeedRepository` is supposed to have a different lifecycle than `FeedViewModel`. If we will add more `Fragments` to the app we don't want to recreate the repository for each screen. 

In other words we want to inject `FeedRepository` into  `FeedViewModel`. And the simplest way to do it is probably using `FeedViewModelFactory`.